zookeeper-server/zookeeper-server-3.5.6/src/test/java/com/yahoo/vespa/zookeeper/ReconfigurerTest.java

```diff
@@ -0,0 +1,75 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.zookeeper;
+
+import com.yahoo.cloud.config.ZookeeperServerConfig;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.stream.IntStream;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests dynamic reconfiguration of zookeeper cluster.
+ */
+public class ReconfigurerTest {
+
+    private File cfgFile;
+    private File idFile;
+
+    @Rule
+    public TemporaryFolder folder = new TemporaryFolder();
+
+    @Before
+    public void setup() throws IOException {
+        cfgFile = folder.newFile();
+        idFile = folder.newFile("myid");
+    }
+
+    @Test
+    public void testStartupAndReconfigure() {
+        Reconfigurer reconfigurer = new Reconfigurer();
+        reconfigurer.startOrReconfigure(createConfig(1));
+
+        // Created config has dynamicReconfig set to false
+        assertFalse(reconfigurer.shouldReconfigure(createConfig(2)));
+
+        // Created config has dynamicReconfig set to true
+        assertTrue(reconfigurer.shouldReconfigure(createConfigAllowReconfiguring(2)));
```

Comment:
Consider testing that an equal config does not cause reconfiguration.

Reply:
Good point, added